### ISSUE TITLE
minor a11y: <select> has no visual indicator when tabbed to

### ISSUE THREAD

**Issue by mustafa0x** [simonw/datasette#1771]

**Comment by simonw** [comment]
OK I see what you mean:

https://latest.datasette.io/fixtures/attraction_characteristic

![Animated GIF of the table page hitting tab a bunch - the cog icon highlights and so does the text input but the two select boxes do not](https://user-images.githubusercontent.com/9599/208280176-1e2de671-fe69-43e8-8d62-bf7aa8f4d36e.gif)

**Comment by simonw** [comment]
The border is actually on the div that wraps the select box:

<img width="1171" alt="image" src="https://user-images.githubusercontent.com/9599/208280249-9d9848b5-6e33-46c3-ad99-bc1948e08f25.png">

I tried adding a `border: 1px dotted black` to `select:focus` but it's not quite right - it jumps around a bit like this:

![Tabbing to the selects shows a 1px border but the element expands in size by one pixel, causing a visual jump](https://user-images.githubusercontent.com/9599/208280271-41a07f68-b8b1-4908-a4e2-aac4304d6c09.gif)

**Comment by mustafa0x** [comment]
the problem is:
```
.select-wrapper select:focus {
  outline: none;
}
```

I sometimes add this js:
```
window.addEventListener('keydown', function check_tab(e) {
    if (e.key === 'Tab') {
        document.documentElement.classList.add('user-is-tabbing')
        window.removeEventListener('keydown', check_tab)
    }
})
```

and then in the css, using a `html.user-is-tabbing` selector undo any outlines I removed.

**Comment by simonw** [comment]
This does the trick:

```css
div.select-wrapper:focus-within {
  border: 1px solid black;
}
```
![tab-select-border-fix](https://user-images.githubusercontent.com/9599/208283826-de48212f-a213-40fc-9b37-9d66f0858f21.gif)

**Comment by simonw** [comment]
Now live on https://latest.datasette.io/fixtures/attraction_characteristic
